easypost/services/beta_referral_customer_service.py

```diff
@@ -36,33 +37,56 @@ def add_payment_method(
         response = Requestor(self._client).request(
             method=RequestMethod.POST,
             url="/referral_customers/payment_method",
-            params=wrapped_params,
+            params=params,
             beta=True,
         )
 
         return convert_to_easypost_object(response=response)
 
     def refund_by_amount(self, refund_amount: int) -> Dict[str, Any]:
         """Refund a ReferralCustomer wallet by specifying an amount."""
-        wrapped_params = {"refund_amount": refund_amount}
+        params = {"refund_amount": refund_amount}
 
         response = Requestor(self._client).request(
             method=RequestMethod.POST,
             url="/referral_customers/refunds",
-            params=wrapped_params,
+            params=params,
             beta=True,
         )
 
         return convert_to_easypost_object(response=response)
 
     def refund_by_payment_log(self, payment_log_id: str) -> Dict[str, Any]:
         """Refund a ReferralCustomer wallet by specifying a payment log ID to completely refund."""
-        wrapped_params = {"payment_log_id": payment_log_id}
+        params = {"payment_log_id": payment_log_id}
 
         response = Requestor(self._client).request(
             method=RequestMethod.POST,
             url="/referral_customers/refunds",
-            params=wrapped_params,
+            params=params,
+            beta=True,
+        )
+
+        return convert_to_easypost_object(response=response)
+
+    def retrieve_credit_card_client_secret(self) -> Dict[str, Any]:
+        """Retrieves a client secret to use with Stripe when adding a credit card."""
+        response = Requestor(self._client).request(
+            method=RequestMethod.POST,
```

Comment:
An empty POST to get something from the API...

Reply:
Super cool huh? 😉 We could call this `create_` instead of `retrieve_` which may make more sense? It is indeed a POST

Reply:
I think it depends on how the API docs content is written. If the API docs state "retrieve a client secret," then the current function naming makes more sense.